File: spec/python/test_process_xor4_value.py
import unittest

from process_xor4_value import ProcessXor4Value

class TestProcessXor4Value(unittest.TestCase):
    def test_process_xor4_value(self):
        with ProcessXor4Value.from_file("src/process_xor_4.bin") as r:

            self.assertEqual(r.key, bytearray([0xec, 0xbb, 0xa3, 0x14]))
            self.assertEqual(r.buf, b"foo bar")

File: spec/lua/test_process_xor4_value.lua
local luaunit = require("luaunit")

require("process_xor4_value")

TestProcessXor4Value = {}

function TestProcessXor4Value:test_process_xor4_value()
    local r = ProcessXor4Value:from_file("src/process_xor_4.bin")

    luaunit.assertEquals(r.key, "\xec\xbb\xa3\x14")
    luaunit.assertEquals(r.buf, "foo bar")
end
